src/main/java/frc/robot/subsystems/shooter/Shooter.java

```diff
@@ -0,0 +1,86 @@
+package frc.robot.subsystems.shooter;
+
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonFX;
+import edu.wpi.first.math.Matrix;
+import edu.wpi.first.math.Nat;
+import edu.wpi.first.math.VecBuilder;
+import edu.wpi.first.math.controller.LinearQuadraticRegulator;
+import edu.wpi.first.math.estimator.KalmanFilter;
+import edu.wpi.first.math.numbers.N1;
+import edu.wpi.first.math.system.LinearSystem;
+import edu.wpi.first.math.system.LinearSystemLoop;
+import edu.wpi.first.math.system.plant.DCMotor;
+import edu.wpi.first.math.system.plant.LinearSystemId;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.subsystems.UnitModel;
+import frc.robot.utils.Utils;
+
+import static frc.robot.Constants.LOOP_PERIOD;
+import static frc.robot.Constants.Shooter.*;
+import static frc.robot.Ports.Shooter.*;
+
+public class Shooter extends SubsystemBase {
+    private static final Shooter INSTANCE = new Shooter();
+    private final UnitModel unitModel = new UnitModel(TICKS_PER_METER);
+    private final WPI_TalonFX mainMotor = new WPI_TalonFX(MAIN_MOTOR);
+    private final LinearSystemLoop<N1, N1, N1> linearSystemLoop;
+    private final DCMotor motor = DCMotor.getFalcon500(1);
+
+    private Shooter() {
+        mainMotor.setInverted(MAIN_INVERTED);
+        mainMotor.setSensorPhase(MAIN_SENSOR_PHASE);
+        linearSystemLoop = configStateSpace("Inertia");
+    }
+
+    public static Shooter getINSTANCE() {
```

Comment:
> Eitan, good job! A few notes:
> 
> * The hood is a different subsystem, it shouldn't be included in this PR. Please move the code to another branch (`git cherry-pick` will be useful) and then I will review it.
> * The code lacks documentation.
> * Rename the branch's name to `shooter` to adhere to convention.
> * Make sure to not make changes that aren't relevant to the shooter in this branch.

Hey Katsuv, thank you for reviewing my subsystem. Since you made your review I have made some changes, so could you please resolve the conversations that are irrelevant?

Reply:
Hey,
1) It's Katzuv 🙂 but Dan is better.
2) Could you please comment on every conversation the hash of the relevant commit commit? It will be much easier for me to re-review.
3) Instead of commenting on a random conversation in my review you can just comment on the whole pull request.
Thanks!